File: src/main/java/ca/cutterslade/experiment/gdk/GreeterComponent.java
package ca.cutterslade.experiment.gdk;

import javax.inject.Singleton;

import dagger.Component;
import dagger.grpc.server.NettyServerModule;
import io.grpc.Server;

@Singleton
@Component(modules = {
    NettyServerModule.class,
    GreeterUnscopedGrpcServiceModule.class,
    GreeterModule.class
})
public interface GreeterComponent extends GreeterServiceDefinition {
  Server server();
}

File: src/main/kotlin/ca/cutterslade/experiment/gdk/GreeterComponent.kt
package ca.cutterslade.experiment.gdk

import dagger.Component
import dagger.grpc.server.NettyServerModule
import io.grpc.Server
import javax.inject.Singleton

@Singleton
@Component(modules = [
  NettyServerModule::class,
  GreeterUnscopedGrpcServiceModule::class,
  GreeterModule::class
])
interface GreeterComponent : GreeterServiceDefinition {
  fun server(): Server
}
